test/specs/palindrom.errors.test.js

```diff
@@ -1,331 +1,291 @@
-        });
-        const spy = sinon.spy();
-        const palindrom = new Palindrom({
-          remoteUrl: 'http://localhost/testURL',
-          onIncomingPatchValidationError: spy
-        });
-        setTimeout(() => {
-          assert(spy.calledOnce);
-          const errorPassed = spy.getCall(0).args[0];
-          assert(errorPassed instanceof RangeError);
-          assert.equal(
-            errorPassed.message,
-            `A number that is either bigger than Number.MAX_INTEGER_VALUE or smaller than Number.MIN_INTEGER_VALUE has been encountered in a patch, value is: ${Number.MAX_SAFE_INTEGER +
-              1}, variable path is: /value`
-          );
-          done();
-        }, 10);
-      });
-      it('Outgoing HTTP patches: out of range numbers should call onOutgoingPatchValidationError with a RangeError', done => {
-        moxios.stubRequest('http://localhost/testURL', {
-          status: 200,
-          headers: { Location: 'http://localhost/testURL' },
-          responseText: `{"val": 1}`
+            });
         });
+        context('Numbers Validation', function() {
+            it('Initial HTTP response: out of range numbers should call onIncomingPatchValidationError event with a RangeError', async () => {
+                fetchMock.mock(getTestURL('testURL'), {
+                    status: 200,
+                    body: `{"value": ${Number.MAX_SAFE_INTEGER + 1}}`
+                });
+                const spy = sinon.spy();
+                const palindrom = new Palindrom({
+                    remoteUrl: getTestURL('testURL'),
+                    onIncomingPatchValidationError: spy
+                });
+
+                await sleep();
+                assert(spy.calledOnce);
+                const errorPassed = spy.getCall(0).args[0];
+                assert(errorPassed instanceof RangeError);
+                assert.equal(
+                    errorPassed.message,
+                    `A number that is either bigger than Number.MAX_INTEGER_VALUE or smaller than Number.MIN_INTEGER_VALUE has been encountered in a patch, value is: ${Number.MAX_SAFE_INTEGER +
+                        1}, variable path is: /value`
+                );
+            });
+            it('Outgoing HTTP patch: out of range numbers should call onOutgoingPatchValidationError event with a RangeError', async () => {
+                fetchMock.mock(getTestURL('testURL-range'), {
+                    status: 200,
+                    body: `{"val": 1}`
+                });
+
+                const spy = sinon.spy();
+
+                new Palindrom({
+                    remoteUrl: getTestURL('testURL-range'),
+                    onStateReset: obj =>
+                        obj.val = Number.MAX_SAFE_INTEGER + 1
+                    ,
+                    onOutgoingPatchValidationError: spy
+                });
+
+                await sleep(20);
+
+                assert(spy.calledOnce, spy.callCount);
```

Comment:
I think It should be `assert(spy.calledOnce);` Otherwise it compares `true` to `1` or `false` to `0,2,...` which is strange.